x-pack/plugins/security_solution/public/detections/components/alerts_table/default_config.tsx

```diff
@@ -317,6 +321,15 @@ export const getAlertActions = ({
     return module === 'endpoint' && kind === 'alert';
   };
 
+  const isFromValidRule = () => {
+    const ruleTypes = getMappedNonEcsValue({
+      data: nonEcsRowData,
+      fieldName: 'signal.rule.type',
+    });
+    const [ruleType] = ruleTypes as Array<Rule['type']>;
```

Comment:
weird typecheck work around, better suggestions welcome

Reply:
`RuleType` is a more direct type reference, but up to you on `Rule['type']` vs `RuleType`. 

If `getMappedNonEcsValue` can return something other than a `string[]` maybe it's worth making it generic (to allow `getMappedNonEcsValue<RuleType[]>`, otherwise I'd say that refining our string to our enum is probably the best we can do here, as far as I'm aware.

Alternately we could loosen the restrictions on those predicate functions and just have them accept strings instead of `RuleType`s, but that would have some (IMO) negative downstream consequences.